RUFAS/routines/field/crop/nitrogen_incorporation.py

```diff
@@ -481,24 +498,64 @@ def _determine_shape_log(
             or emergence_nitrogen_fraction < 0
             or emergence_nitrogen_fraction > 1
         ):
+            om.add_error(
+                "Invalid nitrogen_fraction, heat_fraction, mature_nitrogen_fraction,"
+                " or emergence_nitrogen_fraction.",
+                f"One or more of the following fractions are not between 0 and 1,"
+                f" nitrogen_fraction: {nitrogen_fraction}, heat_fraction: {heat_fraction},"
+                f" mature_nitrogen_fraction: {mature_nitrogen_fraction},"
+                f" emergence_nitrogen_fraction: {emergence_nitrogen_fraction}.",
```

Comment:
```suggestion
                "Received invalid fractional value",
                f"All following values must be in the range (0, 1), received {nitrogen_fraction=}, {heat_fraction=}, {mature_nitrogen_fraction=}, {emergence_nitrogen_fraction=}.",
```
I suggest trying to shorten this message a little, I think it helps make the code more readable.